catalog/dags/es/create_new_es_index/create_new_es_index_types.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass, field
+from datetime import timedelta
+
+from airflow.models.connection import Connection
+from es.recreate_staging_index.recreate_full_staging_index import (
+    DAG_ID as RECREATE_STAGING_INDEX_DAG_ID,
+)
+
+from common.constants import PRODUCTION, STAGING
+from data_refresh.data_refresh_types import DATA_REFRESH_CONFIGS
+from database.staging_database_restore.constants import (
+    DAG_ID as STAGING_DB_RESTORE_DAG_ID,
+)
+
+
+@dataclass
+class CreateNewIndex:
+    """
+    Configuration object for the create_new_es_index DAG.
+
+    Required Constructor Arguments:
+
+    environment:     str representation of the environment in which to create
+                     the new index
+    blocking_dags:   list of dags with which to prevent concurrency; the
+                     generated create_new_es_index dag will fail immediately if
+                     any of these dags are running.
+    reindex_timeout: timedelta expressing maximum amount of time the reindexing
+                     step may take
+    """
+
+    dag_id: str = field(init=False)
+    es_host: str = field(init=False)
+    environment: str
+    blocking_dags: list
+    reindex_timeout: timedelta = timedelta(hours=12)
+
+    def __post_init__(self):
+        self.dag_id = f"create_new_{self.environment}_es_index"
+
+        # Get the appropriate connection information for this environment.
+        conn = Connection.get_connection_from_secrets(
+            f"elasticsearch_http_{self.environment}"
+        )
+        self.es_host = conn.host
```

Comment:
I played around with this so many different ways -- it was tricky because the DAGs are generated per-environment 🥲 I thought this wouldn't be a big issue, but if there are errors using the CLI I don't think we should go with this implementation after all. I've just moved this into an additional Airflow task to get the host, which should be fine!

Reply:
Ahh, that's a perfect way to do it even if it feels a little much!